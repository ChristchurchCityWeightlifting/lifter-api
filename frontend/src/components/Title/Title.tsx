/** @format */

import React from "react";
import { Typography } from "@mui/material";

interface TitleProps {
  children?: React.ReactNode;
}

const Title: React.FC<React.PropsWithChildren<TitleProps>> = (
  props: TitleProps
) => {
  return (
    <Typography
      component="h2"
      color="primary"
      variant="h4"
      gutterBottom
      {...props}
    >
      {props.children}
    </Typography>
  );
};

export default Title;
