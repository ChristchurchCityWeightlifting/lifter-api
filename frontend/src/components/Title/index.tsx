/** @format */

export { default } from "./Title";
