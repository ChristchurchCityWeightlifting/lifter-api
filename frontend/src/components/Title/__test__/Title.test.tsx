/** @format */

import React from "react";
import { describe, test, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import Title from "../Title";

describe("Title component tests", () => {
  test("renders without text", () => {
    const tree = renderer.create(<Title></Title>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text", () => {
    const tree = renderer.create(<Title>Test Text</Title>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text and props", () => {
    const tree = renderer
      .create(<Title sx={{ textAlign: "left" }}>Test Text</Title>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
