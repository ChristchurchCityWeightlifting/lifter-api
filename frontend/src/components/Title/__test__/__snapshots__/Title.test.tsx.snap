// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component tests renders with text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-kyj2s4-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:300;font-size:3.75rem;line-height:1.2;letter-spacing:-0.00833em;margin-bottom:0.35em;color:#1976d2;}",
      }
    }
    data-emotion="css kyj2s4-MuiTypography-root"
  />,
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-kyj2s4-MuiTypography-root"
  >
    Test Text
  </h2>,
]
`;

exports[`Title component tests renders with text and props 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-2n0jr6-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:300;font-size:3.75rem;line-height:1.2;letter-spacing:-0.00833em;margin-bottom:0.35em;color:#1976d2;text-align:left;}",
      }
    }
    data-emotion="css 2n0jr6-MuiTypography-root"
  />,
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-2n0jr6-MuiTypography-root"
  >
    Test Text
  </h2>,
]
`;

exports[`Title component tests renders without text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-kyj2s4-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:300;font-size:3.75rem;line-height:1.2;letter-spacing:-0.00833em;margin-bottom:0.35em;color:#1976d2;}",
      }
    }
    data-emotion="css kyj2s4-MuiTypography-root"
  />,
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-kyj2s4-MuiTypography-root"
  />,
]
`;
